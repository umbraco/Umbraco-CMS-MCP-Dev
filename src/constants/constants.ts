export const BLANK_UUID = "00000000-0000-0000-0000-000000000000";
export const ROOT_DOCUMENT_TYPE_ID = "a95360e8-ff04-40b1-8f46-7aa4b5983096";
export const CONTENT_DOCUMENT_TYPE_ID = "b871f83c-2395-4894-be0f-5422c1a71e48";
export const Default_Memeber_TYPE_ID = "d59be02f-1df9-4228-aa1e-01917d806cda";
export const TextString_DATA_TYPE_ID = "0cc0eba1-9960-42c9-bf9b-60e150b429ae";
export const IMAGE_MEDIA_TYPE_ID = "cc07b313-0843-4aa8-bbda-871c8da728c8";
export const FOLDER_MEDIA_TYPE_ID = "f38bd2d7-65d0-48e6-95dc-87ce06ec2d3d";
export const EXAMPLE_IMAGE_PATH =
  "/src/umb-management-api/tools/temporary-file/__tests__/helpers/example.jpg";
