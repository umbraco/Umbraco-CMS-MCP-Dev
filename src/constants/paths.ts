import { join } from 'path';

// Use absolute path from current working directory
export const RESOURCES_DIR = join(process.cwd(), 'build', 'resources').replace(/\\/g, '/');