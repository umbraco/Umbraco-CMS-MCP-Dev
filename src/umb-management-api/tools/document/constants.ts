export const UmbracoDocumentPermissions = {
    Create: "Umb.Document.Create",
    Read: "Umb.Document.Read",
    Update: "Umb.Document.Update",
    Delete: "Umb.Document.Delete",
    Publish: "Umb.Document.Publish",
    Unpublish: "Umb.Document.Unpublish",
    Move: "Umb.Document.Move",
    Sort: "Umb.Document.Sort",
    Duplicate: "Umb.Document.Duplicate",
    PublicAccess: "Umb.Document.PublicAccess",
    CultureAndHostnames: "Umb.Document.CultureAndHostnames",
    CreateBlueprint: "Umb.Document.CreateBlueprint",
}