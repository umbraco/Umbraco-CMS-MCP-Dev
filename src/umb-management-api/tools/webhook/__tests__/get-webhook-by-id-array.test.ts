import GetWebhookItemTool from "../get/get-webhook-by-id-array.js";
import { WebhookBuilder } from "./helpers/webhook-builder.js";
import { WebhookTestHelper } from "./helpers/webhook-helper.js";
import { jest } from "@jest/globals";
import { BLANK_UUID } from "@/constants/constants.js";
import {
  CONTENT_DELETED_EVENT,
  CONTENT_PUBLISHED_EVENT,
  TEST_WEBHOOOK_URL,
} from "./webhook-constants.js";

describe("get-webhook-item", () => {
  const TEST_WEBHOOK_NAME = "_Test Item Webhook";
  const TEST_WEBHOOK_NAME_2 = "_Test Item Webhook2";
  let originalConsoleError: typeof console.error;

  // Helper to parse response, handling empty string as empty array
  const parseItems = (text: string) => {
    if (!text || text.trim() === "") return [];
    return JSON.parse(text);
  };

  beforeEach(() => {
    originalConsoleError = console.error;
    console.error = jest.fn();
  });

  afterEach(async () => {
    console.error = originalConsoleError;
    await WebhookTestHelper.cleanup(TEST_WEBHOOK_NAME);
    await WebhookTestHelper.cleanup(TEST_WEBHOOK_NAME_2);
  });

  it("should get no webhooks for empty request", async () => {
    // Get all webhooks
    const result = await GetWebhookItemTool().handler(
      {},
      { signal: new AbortController().signal }
    );
    const items = parseItems(result.content[0].text as string);

    expect(items).toMatchSnapshot();
  });

  it("should get single webhook by ID", async () => {
    // Create a webhook
    const builder = await new WebhookBuilder()
      .withName(TEST_WEBHOOK_NAME)
      .withUrl(TEST_WEBHOOOK_URL)
      .withEvents([CONTENT_PUBLISHED_EVENT])
      .create();

    // Get by ID
    const result = await GetWebhookItemTool().handler(
      { id: [builder.getId()] },
      { signal: new AbortController().signal }
    );
    const items = parseItems(result.content[0].text as string);
    expect(items).toHaveLength(1);
    expect(items[0].name).toBe(TEST_WEBHOOK_NAME);
    // Normalize for snapshot
    items[0].id = BLANK_UUID;
    expect(items).toMatchSnapshot();
  });

  it("should get multiple webhooks by ID", async () => {
    // Create first webhook
    const builder1 = await new WebhookBuilder()
      .withName(TEST_WEBHOOK_NAME)
      .withUrl("https://example.com/webhook1")
      .withEvents([CONTENT_PUBLISHED_EVENT])
      .create();

    // Create second webhook
    const builder2 = await new WebhookBuilder()
      .withName(TEST_WEBHOOK_NAME_2)
      .withUrl("https://example.com/webhook2")
      .withEvents([CONTENT_DELETED_EVENT])
      .create();

    // Get by IDs
    const result = await GetWebhookItemTool().handler(
      {
        id: [builder1.getId(), builder2.getId()],
      },
      { signal: new AbortController().signal }
    );

    const items = parseItems(result.content[0].text as string);
    expect(items).toHaveLength(2);
    expect(items[0].name).toBe(TEST_WEBHOOK_NAME);
    expect(items[1].name).toBe(TEST_WEBHOOK_NAME_2);

    // Normalize for snapshot
    items.forEach((item: any) => {
      item.id = BLANK_UUID;
    });
    expect(items).toMatchSnapshot();
  });
});
