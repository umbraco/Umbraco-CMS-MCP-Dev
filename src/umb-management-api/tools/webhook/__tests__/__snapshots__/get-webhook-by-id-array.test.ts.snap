// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get-webhook-item should get multiple webhooks by ID 1`] = `
[
  {
    "enabled": true,
    "events": "content.published",
    "id": "00000000-0000-0000-0000-000000000000",
    "name": "_Test Item Webhook",
    "types": "",
    "url": "https://example.com/webhook1",
  },
  {
    "enabled": true,
    "events": "content.deleted",
    "id": "00000000-0000-0000-0000-000000000000",
    "name": "_Test Item Webhook2",
    "types": "",
    "url": "https://example.com/webhook2",
  },
]
`;

exports[`get-webhook-item should get no webhooks for empty request 1`] = `[]`;

exports[`get-webhook-item should get single webhook by ID 1`] = `
[
  {
    "enabled": true,
    "events": "content.published",
    "id": "00000000-0000-0000-0000-000000000000",
    "name": "_Test Item Webhook",
    "types": "",
    "url": "https://example.com/webhook",
  },
]
`;
